pkg/specs/pvcs.go

```diff
@@ -176,12 +173,56 @@ func DetectPVCs(
 		result.Dangling = append(result.Dangling, pvc.Name)
 	}
 
+	if !cluster.ShouldCreateWalArchiveVolume() {
+		return result
+	}
+
+	for _, instance := range getInstancesName(podList) {
+		expectedPVCs := getExpectedInstancePVCNames(cluster, instance)
+		var indexOfFoundPVCs []int
+
+		for idx, pvcName := range result.Healthy {
+			if slices.Contains(expectedPVCs, pvcName) {
+				indexOfFoundPVCs = append(indexOfFoundPVCs, idx)
+			}
+		}
+
+		if len(expectedPVCs) == len(indexOfFoundPVCs) {
```

Comment:
We could transform this in a `switch` statement, I bet this will be more readable.